### ISSUE TITLE
Post-defense feedback

### ISSUE THREAD

**Issue by dluman** [Mai1902/SeniorThesis_MaiNguyen#7]
Hi, @Mai1902 -- thanks for your presentation last Tuesday. There is much to recommend your effort and the outcome of your process. While you and I have identified several areas of the project that could be taken much further, we're only really concerned with where you are now. In that light, your second reader has two questions for you to address:

# Q1

In the Ethics Section of page 17, A Potential misuse  is that applicants may be encouraged to lie on their CVs and other application documents by adding keywords for skills which the applicant does not actually have. These keywords fool parsing checks which match and score common skill set vocabulary between the applications and job offers. 

GitHub already has a dashboard that lists the types of programming languages that exist in projects. Other attributes of projects also exist on user's page at GitHub concerning the number of PR's and etc. It is likely that more details will be be added to GitHub's dashboard in time.

Could Lander simply read these skills from public online resources like GitHub to verify the applicant's skills when it pairs their applications with job postings? Would that reduce the chances of user giving misleading information on their job applications (when the skills can just be looked-up online)?

# Q2

Q2: Also in the same Ethics Section, Information Privacy is mentioned. The privacy issue could imply that applicants are able to hide information to fool Lander parsing functions.

Could data from public online resources (i.e, GitHub project commits, statistics and etc) be applied to parsing functions to avoid any misleading personal information from derailing an application process.

On the other hand, is forcing (all) users to reveal lots of information about themselves and their projects a violation of ethics? Could something undesirable come of this practice? Discuss the obligation of having all users release personal information on public websites such as GitHub.

If you are interested in using them for discussion points to her largely un-populated Ethics section, then please add them to her Issue tracker.

**Comment by Mai1902** [comment]
Thank you so much, Professor Luman and Professor Bonham-Carter, for providing me with these invaluable insights, which I will definitely consider as I work on my Ethics section and the implementation of my tool. For the two questions above, my answers are as follows:

Q1: Lander will be able to read all the skills listed in candidates' projects if these projects are listed on the resume, or if the candidate provides a link to their Github repositories on their resume. GitHub API has a built-in method for automatically retrieving all programming languages used by a user from their public repositories. I believe that if this addition is successfully implemented, it will aid Lander in fact-checking the talents that applicants list on their resumes, but it may not entirely help with reducing the number of people lying on their resumes.

In addition, I wish to underline that Lander's end-user is job seekers, not recruiters. This tool assists job seekers by analyzing their resumes and recommending job titles and keywords to include to boost their chances of passing the initial screening. Thus, while it is vital to verify the claimed qualifications or skills of users, this should be performed by a recruiter instead of Lander. On the other hand, GitHub API is currently only able to retrieve the programming languages, not frameworks or libraries. And also, some students do list the languages or frameworks that they studied or use to a certain level on their resume but have yet to use this knowledge to implement a public tool on GitHub. Thus, fact-checking candidates by solely extracting skills on their GitHub, while helpful, may be unfair to the candidates. Even Applicant Tracking System, only extract the skills from candidate resume and evaluate their qualifications from there. The reason is most if not all tech company does require candidates to take an intensive technical assessment once they pass the initial screening round. In this round, the company will test only on the skills that are deemed important to this job title, for example, the Data Analyst position may only test on SQL or Python skills, and candidates can still lie that they know how to use C. In this case, if the candidate is good enough in SQL and Python to pass the test, then the fact that they lie on their C competency will be overlooked by recruiters. This example is just to show that the best way to test if the candidate is qualified for the job or not is through assessment, which will be carried out by the recruiter. 

However, I do wish to include GitHub repository checking as a way to remind candidates that if they want to prove their competency in skills, they should have projects on their public repositories to prove that. This addition will definitely help them in the interview round. 

Q2: Lander works as a job seeker's aid; consequently, hiding information or lying to Lander will reduce the effectiveness of this resource. I do feel that the majority of Lander's customers are desperate job seekers hoping to improve their acceptance rate by identifying problems in their resumes or relevant job titles. Providing Lander with insincere information will therefore prevent them from accomplishing their goal. If individuals have lied on their resumes, their resumes may contain an abundance of keywords that could mislead Lander when the tool attempts to match the candidate to a job. And this action will make it more difficult for candidates to find positions that fit their technical level and skill sets,  which will harm the user rather than help them. Thus, I do not expect that candidates will seek to trick the Lander parsing function, as this is the tool that helps them identify missing keywords and missing skills, as well as match them with an appropriate job. Lander is not the Applicant Tracking System algorithm that candidates must outperform.

In addition, the inclusion of sensitive material such as GitHub repositories, GitHub profile information, and other sensitive information on a resume is totally at the discretion of the candidate. If they are concerned about their privacy, they can opt out of using the application. Lander's source code does not contain recording methods, hence this application cannot and will not store user data. The information provided by the user will be used on the spot to evaluate job matches and missing keywords, and the session will be reset after each use. 

In conclusion, the end-user of Lander is job searchers, and the purpose of this tool is to facilitate the job application process for candidates. Hence, Lander will operate most efficiently if the user offers truthful information; however, users are not compelled to divulge any information.

Please let me know if there are any further questions I can answer or if there are other points that I should add to my thesis. Please let me know if my answers to the following questions are not clear enough. I am aware that my language might be ambiguous at times.
